hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -617,13 +620,45 @@ private void setServerRole(RaftPeerRole currentRole,
     }
   }
 
-  private GroupInfoReply getGroupInfo() throws IOException {
+  public GroupInfoReply getGroupInfo() throws IOException {
     GroupInfoRequest groupInfoRequest = new GroupInfoRequest(clientId,
         raftPeerId, raftGroupId, nextCallId());
     GroupInfoReply groupInfo = server.getGroupInfo(groupInfoRequest);
     return groupInfo;
   }
 
+  public Map<String, String> getServerRoles() throws IOException {
+    GroupInfoReply groupInfo = getGroupInfo();
+    if (groupInfo == null) {
+      throw new IOException(ozoneManager.getOMNodeId() +
+          ": Failed to get GroupInfo.");
+    } else {
+      Map<String, String> serverRoles = new HashMap<>();
+
+      RoleInfoProto roleInfoProto = groupInfo.getRoleInfoProto();
+      String selfNodeId = new String(roleInfoProto.getSelf().getId()
+          .toByteArray(), StandardCharsets.UTF_8);
+      String selfRole = roleInfoProto.getRole().name();
+      serverRoles.put(selfNodeId, selfRole);
+
+      if (roleInfoProto.hasLeaderInfo()) {
+        for (ServerRpcProto followerInfo :
+            roleInfoProto.getLeaderInfo().getFollowerInfoList()) {
+          String nodeId = new String(
+              followerInfo.getId().getId().toByteArray());
```

Comment:
Findbugs violation: Specify encoding scheme.

Reply:
Thanks Dinesh. Fixed it.